pylinalg/func/matrix.py

```diff
@@ -464,3 +464,39 @@ def matrix_make_orthographic(
     out[3, 3] = 1
 
     return out
+
+
+def matrix_make_look_at(eye, target, up, /, *, out=None, dtype=None):
+    """
+    Rotation that aligns two vectors.
+
+    Given a camera (frame) looking in direction ``eye``, this function computes
+    a rotation matrix that makes the camera look into direction ``target``. It
+    then rotates the camera frame around ``target`` to make the camera's up
+    direction point in direction ``up``.
```

Comment:
Having written this, I just realized that the signature for this function is incomplete. We don't know how a camera frame is oriented from a single vector  (`eye`), as the camera could have an arbitrary rotation around `eye`. 

Do we want to add a `old_up` argument to the function, or do we want to remove `up` and make the function perform a pure planar rotation from one vector onto another?

Reply:
This line is incorrect: `Given a camera (frame) looking in direction ``eye``, this function computes a rotation matrix that makes the camera look into direction ``target``.`

The assumption is that the z-axis is the view axis, and you're going to create a rotation such that the z-axis aligns with the line from the point `eye` to the point `target`.

Refer to this implementation: https://github.com/mrdoob/three.js/blob/dev/src/math/Matrix4.js#L283-L328

For "up", we assume that the y-axis is the up axis, and you're going to rotate about the new view axis (from `eye` to `target`) such that the y-axis aligns with `up`.

Reply:
Ah, so it's `(0, 0, 1) -> (target-eye)` and `(0, 1, 0) --> up`. I'll fix the docstring.

Reply:
@Korijn Actually thinking about this again, could we drop `eye` in that signature and just do `target`?

`matrix_make_look_at(target, up, /, *, out=None, dtype=None)` sounds much more crisp, and we can get the current behavior via `matrix_make_look_at(target-eye, up)`, which I think is less surprising than passing in eye and target separately.

Reply:
Sure. You're then assuming eye is (0, 0, 0)?

Reply:
No. At the moment the first step of `look_at` in pygfx is `_tmp_vector.sub_vectors(eye, target)`. I'm proposing to pass in `_tmp_vector` directly instead of passing in `eye` and `target`, because users can easily pass `look_at(target - eye, ...)` to get the previous functionality.

Since `tmp_vector` is an odd name in a signature, I would just name this `target`, because that is the actual direction the camera will look into as a result of the transformation.

Reply:
In that case I think the name `target` is misleading. Commonly lookat functions take a position `target`, not a direction. Let's either stick to the common signature, or clearly signal through different names that our implementation is not as usual.

Reply:
I like `direction`; however, we can also keep it as-is if that means we match the common signature for this type of function. Your call.

Reply:
Personally, I'd prefer to keep `eye` and `target` as positions

Reply:
Okay, then we keep it as it is right now :)